src/service.js

```diff
@@ -24,10 +24,14 @@ import {dev} from './log';
  * - obj: Actual service implementation when available.
  * - promise: Promise for the obj.
  * - resolve: Function to resolve the promise with the object.
+ * - ctor: Constructor for the service.
+ * - factory: Factory to create the service.
  * @typedef {{
  *   obj: (?Object),
  *   promise: (?Promise|undefined),
  *   resolve: (?function(!Object)|undefined),
+ *   ctor: (?function(new:Object, !Window)|?function(new:Object, !./service/ampdoc-impl.AmpDoc)),
+ *   factory: (?function(!Window)|?function(!./service/ampdoc-impl.AmpDoc))
```

Comment:
First one SGTM. The second seems incorrect to me. We want either a function that takes in a window, or a function that takes in an ampdoc. That reads to me as "a function that can take either a window or an ampdoc" which isn't what we want.